yunwoo/programmers-0/영어가-싫어요.js

```diff
@@ -0,0 +1,27 @@
+function solution(numbers) {
+  const arr = [
+    "zero",
+    "one",
+    "two",
+    "three",
+    "four",
+    "five",
+    "six",
+    "seven",
+    "eight",
+    "nine",
+  ];
+  let copyNumbers = numbers;
+
+  for (let i = 0; i < arr.length; i++) {
+    copyNumbers = copyNumbers.replaceAll(arr[i], i);
+  }
+
+  return +copyNumbers;
+}
```

Comment:
```suggestion
function solution(numbers) {
  const numberMappingArr = [
    "zero",
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
    "nine",
  ];
  let copyNumbers = numbers;
  
  numberMappingArr.forEach((str, i) => {
    copyNumbers = copyNumbers.replaceAll(str, i);
  });

  return +copyNumbers;
}
```
- 풀이잘 봤습니다 🙇‍♂️ 단순히 arr 변수명 보다는 number를 매핑해주는 역할을 하는 배열이기 때문에 `numberMappingArr`로 변경하는 것이 좋을 것 같습니다.
- 또한 해당 풀이는 `forEach`를 사용하면 더 깔끔한 풀이가 가능합니다 😄
- 아래 풀이는 윤우님 코드와 큰 차이는 없지만 정규표현식을 이용한 풀이입니다. 참고해주셔도 좋습니다.
```js
// 정규 표현식 풀이
// 정규 표현식 풀이
function solution(numbers) {
  const numberMappingObj = {
    zero: 0,
    one: 1,
    two: 2,
    three: 3,
    four: 4,
    five: 5,
    six: 6,
    seven: 7,
    eight: 8,
    nine: 9,
  };
  const regex = /zero|one|two|three|four|five|six|seven|eight|nine/g;

  return +numbers.replace(regex, (v) => numberMappingObj[v]);
}
```

Reply:
- 헉 변수명을 마지막에 변경을안했네요..😅
- forEach가 있었군요!! 참고하겠습니다ㅎㅎ
- 객체를 이용해도 되는군요!!! 참고풀이 감사합니다🙇‍♂️